packages/ui-react/src/components/CardGrid/CardGrid.tsx

```diff
@@ -58,7 +61,7 @@ function CardGrid({
 }: CardGridProps) {
   const breakpoint: Breakpoint = useBreakpoint();
   const posterAspect = parseAspectRatio(playlist.cardImageAspectRatio || playlist.shelfImageAspectRatio);
-  const visibleTiles = cols[breakpoint] + parseTilesDelta(posterAspect);
+  const visibleTiles: VisibleTiles = (cols[breakpoint] + parseTilesDelta(posterAspect)) as VisibleTiles;
```

Comment:
IMO There is no need for typings here.  Besides, the variable type is already inferred by the value. Which means you can remove the variable type:

```suggestion
  const visibleTiles = (cols[breakpoint] + parseTilesDelta(posterAspect)) as VisibleTiles;
```

Reply:
I do need the typing for the scss class selector, that's why I introduced it.

Reply:
But ... this is actually telling you something else. The problem still exists but is masked by typecasting it to a "satisfied" type. The same applies to the aspect ratio prop.

So perhaps it would be better to validate it in the component like so;

```ts
// 1; check if the key exists in the styles object and safely cast it as a keyof the styles object.
const getColsClassName = (visibleTiles: number) => {
  const key = `cols-${visibleTiles}`;
  return key in styles ? styles[key as keyof typeof styles] : undefined;
}

// 2; validate based on pre-existing cols, we still need to cast to prevent having a second helper function
const cols = Array(10).map(num => `cols-${num}`);
const getColsClassName = (visibleTiles: number) => {
  const key = `cols-${visibleTiles}`;
  return cols.includes(key) ? styles[key as keyof typeof styles] : undefined;
}


// 3; same as 1, but used as generic util
const getOptionalClassName = <T>(styles: T, className: string) => {
  return className in styles ? styles[className as keyof typeof styles] : undefined;
}
```